client/my-sites/email/email-providers-comparison/in-depth/comparison-table.tsx

```diff
@@ -110,10 +112,23 @@ const ComparisonTable = ( {
 					<td></td>
 
 					{ emailProviders.map( ( emailProviderFeatures ) => {
+						return <td key={ emailProviderFeatures.slug }>{ emailProviderFeatures.badge }</td>;
```

Comment:
Two thoughts here:
1. It's probably worth making the key more specific, e.g. `key={ emailProviderFeatures.slug + '-badge' }`, as using just the slug could result in key name collisions in the same React subtree.
2. It may be worth adding a comment that we always want this `td` to be present, even if `emailProviderFeatures.badge` is undefined.

Reply:
> using just the slug could result in key name collisions in the same React subtree.

Are you sure about this? According to the [documentation](https://reactjs.org/docs/lists-and-keys.html):

> Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside the array to give the elements a stable identity
> 
> Keys only make sense in the context of the surrounding array.
>
> Keys used within arrays should be unique among their siblings. However, they don’t need to be globally unique. 

As far I can see, this only applies to the current loop and not the rest of the React tree.

Reply:
> It may be worth adding a comment that we always want this td to be present, even if emailProviderFeatures.badge is undefined.

That seems excessively cautious to me, I think I'll keep it as it is.

Reply:
For the docs reference, I was mostly working off a React error I'd noticed at some point, but never dug into. Definitely a clarifying and helpful, TIL for me! Thanks for pointing out my misunderstanding!

And I am OK with not adding the comment -- that was a "maybe" suggestion.